### ISSUE TITLE
Colon in item lore?

### ISSUE THREAD

**Issue by Xentax5** [FlyingPikachu/Quests#275]
(__Required__) What is your Quests version / build number (do _not_ say "latest")? 2.7.6-b
Hello, this is more a question then issue so. I have my own rpgitems, that have stats like "Damage: 10" in lore and i want to be able to request these items by npcs (and maybe give them and so on but for give i can use command). Problem is, when I want to add that item to objective deliver items, it says this:
"not formatted properly error".
I used the in game editor to set objective like this: 
```
items-to-deliver:
          - 'name-WOOD_AXE:amount-1:displayname-§aUčňova Sekera:lore-§4»Damage: 3-5«:lore-§1»Durability:
            300/300«:lore-§5»Level 1«'
```
I see that colon is used as divider for key-value, but can I use a escape sequence or something?
And maybe this IS an issue, because i used in game editor and it should correctly insert colons in the lore.

Thank you for a reply.

**Comment by Xentax5** [comment]
Heyo,
don't want to be unpatient, but is someone checking it here or is this dead?

**Comment by FlyingPikachu** [comment]
Hello! You correct that, because colons are used as dividers (which is inherent to the format), they cannot currently be used in lore, displaynames, et al.

I would be happy to add the ability to use escape characters, provided it's a one line solution. However, I'm currently involved in reformatting the language files (again), and with MC 1.13 around the corner (which will affect item data), I can't provide any sort of estimate atm as to when I can add this.

**Comment by ScuroK** [comment]
@FlyingPikachu the colons are crucial for many things! latest update breaks alot! sorry to stress you but it really damaged my server gameplay :( could you do something about it pls? maybe use another charakter as dividers? i really have to fix this ASAP

**Comment by SXRWahrheit** [comment]
I mentioned this on another thread about items that Quests can't currently deliver effectively, but you can use a Denizen script combined with Quests to achieve this functionality in the meantime.

**Comment by FlyingPikachu** [comment]
I would like to point out that the “latest update” did NOT break anything, colons have ALWAYS been a config-breaking character for item values and recent commits merely discourage/prevent their use.

@SXRWahrheit has provided a good workaround for the time being.

**Comment by ScuroK** [comment]
im having a hard time to understand your decision. the older version was working perfectly fine on my server!! and now it fcked up everything i created...

**Comment by ScuroK** [comment]
@FlyingPikachu 
there are more problems than timed commands! even simple things like choosing if a command is executed by a certain plugin or minecraft can only be done with colons!
- teleport
-  minecraft:teleport

for example! and there are many problems if we dont have this choice anymore! if teleport is executed by essentials, players will be able to do /back and go back to restricted quest areas for example.

and id DID work fine before!! :(

**Comment by FlyingPikachu** [comment]
@ScuroK For what feels like the hundredth time, older versions of Quests **did not** allow colons in lore or commands. Unless you have evidence to the contrary, please stop asserting that. The issues you've reported have been caused by unsupported colon use (#318) and encoding problems (#323).

Moving on, I said earlier in this ticket that I'll be looking into allowing colons in a future update. Unless you intend to submit a pull request, you'll have to use a workaround for the time being.

**Comment by FlyingPikachu** [comment]
@ScuroK Think I might've found a solution for commands. Could you please post the language file you're currently using (found in the /lang folder)?

**Comment by ScuroK** [comment]
@FlyingPikachu i dont understand why you keep repeating. i know that it worked for me before. and i did it for a long time. so why do you keep telling me i cant use comand evvents like essentials:teleport or similar? i know what im writing: i did use command events with colons without any problems for a looong time untill your vietnamese update came out. instead of feeling stressed you should try to have a little trust in someone who even payed you a coffe before..

here is my current lang file without strings.new.
https://pastebin.com/mZydWzpS

**Comment by SXRWahrheit** [comment]
@ScuroK As a matter of course, it's incredibly rude to say "I donated to you, therefore you should do what I want." A donation doesn't come with strings attached.

Maybe your old commands happened to work, but if they did, it was *unsupported and unintentional.*

Pikachu might be able to find a fix for you, but in the meantime, you should probably use Denizen for anything not explicitly supported by Quests. It works very well.

**Comment by FlyingPikachu** [comment]
@ScuroK I keep having to repeat myself because you have. Here, and [here](https://github.com/FlyingPikachu/Quests/issues/309), and [here](https://github.com/FlyingPikachu/Quests/issues/318), and [here](https://github.com/FlyingPikachu/Quests/issues/320), and [here.](https://github.com/FlyingPikachu/Quests/issues/326) Oh, and [here](https://github.com/FlyingPikachu/Quests/commit/ca117da950f9fb82c713bbf9136ccb4f7f68b7e8).

You've opened 2 issues and replied to 4 others. Of the latter 4, one was entirely unrelated and another I had to delete your comment because the ticket was made _over a year ago_. If you hadn't had a genuine issue, this would be considered spam. I understand your frustration, but as the only person contributing code at the moment and having seen a lot of user-caused problems (which you've already shown to be prone to in #318 and #323), understand that I must take anything _anyone_ says with a grain of salt. It's far more likely that some form of innocent mistake or neglect caused the issue you're having.

I will now prove that to be the case:

The [strings.yml you provided](https://pastebin.com/mZydWzpS) is _missing_ a lot of strings that were introduced in recent versions of the plugin. Most importantly, it's missing the `charSemi` string introduced in version 2.3.4 (that's PRIOR to the Vietnamese update). Did you miss the bold, red **BREAKING** label I put? Thought you might've seen it, since you Liked that update...

![finito](https://user-images.githubusercontent.com/2267126/36955562-2fb7ed48-1ff7-11e8-9d28-f6e2c8c9c6c9.png)

Now, I have to admit a mistake I made previously in saying that "colons have ALWAYS been a config-breaking character". What I _should_ have said was "_separators_ have ALWAYS been a config-breaking character". This is because I forgot that Quests started using the value of `charSemi` as the separator as of 2.3.4. _The code still handles separators the same as previous versions._ The only difference is that `charSemi` is used instead of colons.

So what does this mean? It means both lore _and_ commands have allowed the use of whatever separator you like as 2.3.4 by setting the value of `charSemi`. If you set it to colons, colons will not be allowed just like all previous versions. If you set it to semicolons (the new default value that would have generated if you followed the BREAKING instructions), then semicolons would not be allowed instead.

**Tl;dr:** [Download the latest dev build](http://ci.ac3-servers.eu/job/Quests/271/) in which I removed the colon restriction, and either regenerate your strings.yml or add in the `charSemi` line from strings_new.yml (if doing this to any language other than English, make sure to save in UTF-8)

Please don't think I'm upset with you, as I know all too well that mistakes happen. Do keep in mind that @SXRWahrheit is right in saying that, although coffee is appreciated, it does not entitle you to get upset over a _free_ plugin. Perhaps I should've closed this ticket as I bet OP is loving all the email notifications. That said, I'm closing it now as @Xentax5 can just update and change his `charSemi` value to use a separator other than colons. Cheers!

**Comment by ScuroK** [comment]
thanks for your answer @FlyingPikachu 

im glad i was not the only one who did mistakes.

btw i didnt get upset because i donated. i simply got frustrated because my quests got fucked up again. thats nothing personal.
